includes/Core/Authentication/Google_Proxy.php

```diff
@@ -399,24 +450,59 @@ public function unregister_site( Credentials $credentials ) {
 		return $this->request( self::OAUTH2_DELETE_SITE_URI, $credentials );
 	}
 
+	/**
+	 * Registers the site on the proxy.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string|WP_Error Redirect URL on success, otherwise an error.
+	 */
+	public function register_site() {
+		return $this->sync_site_fields( null, null );
+	}
+
 	/**
 	 * Synchronizes site fields with the proxy.
 	 *
 	 * @since 1.5.0
+	 * @since n.e.x.t Updated the function to return redirect URL.
 	 *
 	 * @param Credentials $credentials Credentials instance.
 	 * @param string      $mode        Sync mode.
-	 * @return array|WP_Error Response of the wp_remote_post request.
+	 * @return string|WP_Error Redirect URL on success, otherwise an error.
 	 */
-	public function sync_site_fields( Credentials $credentials, $mode = 'async' ) {
-		return $this->request(
+	public function sync_site_fields( Credentials $credentials = null, $mode = 'async' ) {
```

Comment:
This method should _always_ be called with credentials being passed, to sync site fields they need to be present.

I like your idea of reusing the same logic, but could we do it in a way that both public methods "make sense" by themselves? How about we introduce a private method with the parts that is common between them which can be controlled with whatever parameters are needed, but the public method should "suggest" the right parameters that are needed.

Basically this method's parameters should remain like they were before, with `Credentials` instance being required.

Reply:
Yep, makes sense. Updated.